Sources/TuistCloud/Client/AuthenticationMiddleware.swift

```diff
@@ -0,0 +1,33 @@
+import OpenAPIRuntime
+import Foundation
+import TuistSupport
+
+/// Injects an authorization header to every request.
+struct AuthenticationMiddleware: ClientMiddleware {
+    func intercept(
+        _ request: Request,
+        baseURL: URL,
+        operationID: String,
+        next: (Request, URL) async throws -> Response
+    ) async throws -> Response {
+        var request = request
+        let environment = ProcessInfo.processInfo.environment
+        let tokenFromEnvironment = environment[Constants.EnvironmentVariables.cloudToken]
+        let token: String?
+        if CIChecker().isCI() {
+            token = tokenFromEnvironment
+        } else {
+            token = try? tokenFromEnvironment ?? CredentialsStore().read(serverURL: baseURL)?.token
+        }
```

Comment:
Is this the only place where this is being done? I was wondering if there's an opportunity to move this down to `CredentialsStore()`.read..`

Reply:
Reused the `CloudAuthenticationController` helper class that does this 👍